src/chainparams.cpp

```diff
@@ -231,7 +232,7 @@ class CTestNetParams : public CChainParams {
         fDefaultConsistencyChecks = false;
         fRequireStandard = false;
         m_is_test_chain = true;
-
+        m_is_mockable_chain = false;
```

Comment:
In commit 3392d690cfb338e1a9a9d7347710c0744e4486d5:

this should be true, otherwise one has to change this in the source code when wanting to test on testnet. While there is an argument for test params to be as close to main params as possible, I think this does not apply to params that have been introduced for primarily testing purposes. (See also my `m_is_test_chain`, which turned out to be too general and people prefer a per-parameter toggle).

So I think this should be true to not hinder testing on testnet.

Reply:
hmmm, well if we want `m_is_mockable_chain` to be true on testnet, wouldn't it be better not to introduce the param and use the `m_is_test_chain` instead?

Reply:
Although, conceptually, I don't know if I'm convinced. Mocking time seems useful for increasing automated test coverage, but running the node on testnet simulating closer to mainnet makes sense. But I feel pretty indifferent since even if time is allowed to be mocked on testnet, a user doesn't have to invoke it. And there's no concern around attack because its testnet. Mostly just trying to move the PR forward.

Reply:
Feel free to ignore, this can be handled in the follow-up

Reply:
not addressed, but noted as open question in the follow up https://github.com/amitiuttarwar/bitcoin/pull/2#issuecomment-584885826. going to resolve here.